rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,116 @@
+import inspect
+from sanic import Blueprint, response
+from sanic.request import Request
+from sanic.response import HTTPResponse
+from twilio.twiml.voice_response import VoiceResponse, Gather
+from typing import Text, Callable, Awaitable, List
+
+from rasa.shared.core.events import BotUttered
+from rasa.core.channels.channel import (
+    InputChannel,
+    CollectingOutputChannel,
+    UserMessage,
+)
+
+
+class TwilioVoiceInput(InputChannel):
+    @classmethod
+    def name(cls) -> Text:
+        """Name of your custom channel."""
+        return "twilio_voice"
+
+    def blueprint(
+        self, on_new_message: Callable[[UserMessage], Awaitable[None]]
+    ) -> Blueprint:
+
+        twilio_voice_webhook = Blueprint(
+            "custom_webhook_{}".format(type(self).__name__),
+            inspect.getmodule(self).__name__,
+        )
+
+        @twilio_voice_webhook.route("/", methods=["GET"])
+        async def health(request: Request) -> HTTPResponse:
+            return response.json({"status": "ok"})
+
+        @twilio_voice_webhook.route("/webhook", methods=["POST"])
+        async def receive(request: Request):
+            sender_id = request.form.get("From")  # method to get sender_id
+            text = request.form.get("SpeechResult")  # method to fetch text
+            input_channel = self.name()  # method to fetch input channel
+            call_status = request.form.get("CallStatus")  # method to fetch call status
+            call_sid = request.form.get("CallSid")  # Get the call identifier
```

Comment:
To be honest I don't see the beneft of the comments as I think the code is already concise enough (maybe except for `call_sid`). I'd dump the comments or replace them with more useful ones.